src/topology/noetherian_space.lean

```diff
@@ -0,0 +1,197 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.sets.compacts
+import order.order_iso_nat
+import order.compactly_generated
+
+/-!
+# Noetherian space
+
+A Noetherian space is a topological space that satisfies any of the following equivalent conditions:
+- `well_founded ((>) : opens α → opens α → Prop)`
+- `well_founded ((<) : closeds α → closeds α → Prop)`
+- `∀ s : set α, is_compact s`
+- `∀ s : opens α, is_compact s`
+
+The first is chosen as the definition, and the equivalence is shown in
+`topological_space.noetherian_space_tfae`.
+
+## Main Results
+- `noetherian_space.set`: The subspaces of a noetherian space is noetherian.
+- `noetherian_space.is_compact`: Every subset of a noetherian space is compact.
+- `noetherian_space_tfae`: Describes the equivalent definitions of noetherian spaces.
+- `noetherian_space.range`: The image of a noetherian space is noetherian.
+- `noetherian_space.discrete`: A noetherian and hausdorff space is discrete.
+- `noetherian_space.exists_finset_irreducible` : Every closed subset of a noetherian space is
+- `noetherian_space.finite_irreducible_components `: The irreducible components of a noetherian
+  space is finite.
+
+-/
+
+variables (α β : Type*) [topological_space α] [topological_space β]
+
+namespace topological_space
+
+/-- Type class for noetherian spaces. It is defined to be spaces whose open sets satisfies ACC. -/
+@[mk_iff]
+class noetherian_space : Prop :=
```

Comment:
`is_noetherian` is taken (although in a different namespace) for noetherian modules, and I think it would be better if the names are distinct enough.

Reply:
Another alternative is to move it out from the namespace, just like `compact_space` and `t2_space` etc.

Reply:
Sounds good. `is_noetherian` and `is_noetherian_ring` are already in the root namespace, and dot notations are impossible for all three. And if we follows the topological space convention, then it's `noetherian_space` instead of `is_noetherian_space`.